File: 11vaFeriaAcademicayCultural/Intermedio/Problema1/problema1.cpp
/*Problema 1 Cesar Secret
Propuesta de solucion
autor: Efrain Esau Martinez Macias
C++*/
#include <iostream>
#include <string.h>
using namespace std;
int main(int argc, char const *argv[]){
	int clave, n; // enteros para el valor de clave de encriptado y numero de casos
	char mensaje[100]; // arreglo de chars para el mensaje
	cin>>n;
	for(int j = 0; j < n; j++){
		cin>>mensaje;
		clave=3;
		//Desencriptado
		char Aux[150];
		for(int i = 0; i <= strlen(mensaje)-1; i++){
			Aux[i]=mensaje[i]-clave; // le restamos 3 al valor ascii del mensaje encriptado
			if (Aux[i] < 'a') Aux[i]+=26;// si el valor es menor a 'a' le aumentamos 26 para convertirlo en letra
			if (Aux[i] < 'a' or Aux[i] > 'z') Aux[i]=' ';// si no es una letra, significa que es un espacio
		}
		Aux[strlen(mensaje)] = '\0';//agregarmos un fin de linea para evitar imprimir valores basura
		cout<<Aux<<endl;
	}
	return 0;
}

File: 11vaFeriaAcademicayCultural/Intermedio/Problema1/problema1.java
/*Problema 1 Cesar Secret
Propuesta de solucion
autor: Efrain Esau Martinez Macias
Java*/
import java.util.Scanner;

class Problema1{
  public static void main(String[] args) {
    int clave = 3;
    Scanner entrada = new Scanner(System.in);
    int n = entrada.nextInt();
    entrada.nextLine();
    for(int i = 0; i < n; i++){
      String mensaje = entrada.nextLine();
      char [] Aux = new char[150];
      for(int j = 0; j < mensaje.length(); j++){
        //obtenemos el valor ascii del caracter y le restamos la clave oara desencriptar
        int valor = mensaje.codePointAt(j) - clave;
        Aux[j] = (char)valor;//convertimos el valor ascii a char
        if (Aux[j] < 'a') Aux[j]+=26;// si el valor es menor a 'a' le aumentamos 26 para convertirlo en letra
    		if (Aux[j] < 'a' || Aux[j] > 'z') Aux[j]=' ';// si no es una letra, significa que es un espacio
      }
      Aux[mensaje.length()] = '\0';//agregarmos un fin de linea para evitar imprimir valores basura
      System.out.println(Aux);
    }
  }
}
